src/main/java/Duke.java

```diff
@@ -1,10 +1,359 @@
+                    break;
+                }
+                Task newEvent = new Event(commandSubject, timeline);
+                taskList.add(newEvent);
+                displayToDoWithTime(lineLogo, commandSubject, taskIncomplete, taskList.size(),
+                        'E', timeline);
+                writeToFile(taskList);
+                break;
+
+            case "done":
+                taskDone(lineLogo, commandSubject, taskList, taskComplete);
+                writeToFile(taskList);
+                break;
+
+            case "list":
+                taskList(lineLogo, taskList, taskComplete, taskIncomplete);
+                break;
+
+            case "delete":
+                taskDelete(lineLogo, commandSubject, taskList);
+                break;
+
+            default:
+                throw new DukeException("Sorry! I can't find your command. Please retype your command.");
+
+            }
+        } catch(DukeException e){
+            System.out.println(lineLogo);
+            System.out.println(e.getMessage());
+            System.out.println(lineLogo);
+
+        } catch(NumberFormatException e){
+            System.out.println(lineLogo);
+            System.out.println("Expected integer. Please enter a number.");
+            System.out.println(lineLogo);
+        }
+
+}
+
+    private static void taskDelete(String lineLogo, String commandSubject, ArrayList<Task> taskList) throws IOException {
+        int taskNum = Integer.parseInt(commandSubject);
+        taskNum = taskNum - 1;
+        char taskType = taskList.get(taskNum).getTaskType();
+        String taskStatus = (taskList.get(taskNum).isTaskComplete())
+                ? taskComplete : taskIncomplete;
+        if(taskType == 'T'){
+            displayDeleteTodo(lineLogo, taskList.get(taskNum).getTaskName(),
+                    taskStatus, taskList.size() - 1, taskType);
+        }else{
+            displayDeletedTodoWithTime(lineLogo, taskList.get(taskNum).getTaskName(),
+                    taskStatus, taskList.size() - 1, taskType, taskList.get(taskNum).getTimeline());
+        }
+        taskList.remove(taskNum);
+        writeToFile(taskList);
+    }
+
+
+    public static void taskList(String lineLogo, ArrayList<Task> taskList, String taskComplete,
+            String taskIncomplete) {
+        System.out.println(lineLogo);
+        if (taskList.size() == 0){
+            System.out.println("There are no tasks left for you. Add a new task!");
+        }
+        else{
+            System.out.println("Here are the tasks in your list:");
+        }
+        for(int i = 0; i < taskList.size(); i++){
+            String taskStatus = (taskList.get(i).isTaskComplete())
+                    ? taskComplete : taskIncomplete;
+            taskPrinter(taskList, i, taskStatus);
+        }
+        System.out.println(lineLogo);
+    }
+
+    public static void taskPrinter(ArrayList<Task> taskList, int i, String taskStatus) {
```

Comment:
Try to name methods with verbs describing what the method is doing. In this case, maybe taskPrinter can be renamed to printTask?